docs/basics/connection-approval.md

```diff
@@ -61,13 +62,64 @@ private void ApprovalCheck(NetworkManager.ConnectionApprovalRequest request, Net
+    response.Reason = "Some reason for not approving the client";
 
     // If additional approval steps are needed, set this to true until the additional steps are complete
     // once it transitions from true to false the connection approval response will be processed.
     response.Pending = false;
 }
 ```
 
+## Sending An Approval Declined Reason (_`NetworkManager.ConnectionApprovalResponse.Reason`_)
+
+Under the condition that you need to deny a player from connecting for any particular reason (*reached maximum number of connections, invalid authorization, etc.*), the `NetworkManager.ConnectionApprovalResponse` structure provides you with the optional `NetworkManager.ConnectionApprovalResponse.Reason` property.  When `NetworkManager.ConnectionApprovalResponse.Approved` is false and `NetworkManager.ConnectionApprovalResponse.Reason` has been populated with the reason for denying the player's request to connect, then the server will send the client a `DisconnectReasonMessage`. Upon the client side receiving the `DisconnectReasonMessage`, the `NetworkManager.DisconnectReason` property will be populated with the `NetworkManager.ConnectionApprovalResponse.Reason` message. The example provided below demonstrates how this works:
+
+
+```csharp
+using UnityEngine;
+using Unity.Netcode;
+
+/// <summary>
+/// Connection Approval Handler Component
+/// </summary>
+/// <remarks>
+/// This should be placed on the same GameObject as the NetworkManager.
+/// It automatically declines the client connection for example purposes.
+/// </remarks>
+public class ConnectionApprovalHandler : MonoBehaviour
+{
+    private NetworkManager m_NetworkManager;
+
+    private void Start()
+    {
+        m_NetworkManager = GetComponent<NetworkManager>();
+        if (m_NetworkManager != null)
+        {
+            m_NetworkManager.OnClientDisconnectCallback += OnClientDisconnectCallback;
+            m_NetworkManager.ConnectionApprovalCallback = ApprovalCheck;
+        }
+    }
+
+    private void ApprovalCheck(NetworkManager.ConnectionApprovalRequest request, NetworkManager.ConnectionApprovalResponse response)
+    {
+        response.Approved = false;
+        response.Reason = "Testing the declined approval message";
+    }
+
+    private void OnClientDisconnectCallback(ulong obj)
+    {
+        if (!m_NetworkManager.IsServer && m_NetworkManager.DisconnectReason != string.Empty)
+        {
+            Debug.Log($"Approval Declined Reason: {m_NetworkManager.DisconnectReason}");
+        }
+    }
+}
+```
+
+
 ## Connection data  (_`NetworkManager.ConnectionApprovalRequest.Payload`_)
 
 The `ConnectionApprovalRequest.Payload` parameter takes any custom data of your choice that the client should send to the server. Usually, this data should be some sort of ticket, room password, or similar that will decide if a connection should be approved or not. The `connectionData` is specified on the Client-side in the `NetworkingConfig` supplied when connecting.
```

Comment:
The `ConnectionApprovalRequest.Payload` parameter takes any custom data the client should send to the server. The data is usually a matchmaking ticket, room password, or something similar that determines if a connection should be approved or not. The `connectionData` is specified on the client-side in the `NetworkingConfig` supplied when connecting.